rrrspec-client/lib/rrrspec/client/rspec_runner_factory.rb

```diff
@@ -0,0 +1,24 @@
+require 'rrrspec/client/rspec_runner'
+
+module RRRSpec
+  module Client
+    class RSpecRunnerFactory
+      attr_reader :next_factory
+
+      def initialize(successor)
+        @next_factory = successor
+      end
+
+      def create(file_extension)
+        unless Thread.current[:rspec3_runner]
+          Thread.current[:rspec3_runner] = RSpecRunner.new
```

Comment:
Why can you assume that the Runners are thread-level initialization, not process-level? I think the contract in RSpecRunner is that it does the process-level initialization.